sqlmesh/core/model/definition.py

```diff
@@ -2749,23 +2750,37 @@ def render_field_value(value: t.Any) -> t.Any:
 
         return value
 
+    def parse_strings_with_macro_refs(value: t.Any) -> t.Any:
+        if isinstance(value, str) and "@" in value:
+            return exp.maybe_parse(value, dialect=dialect)
+
+        if isinstance(value, dict):
+            for k, v in dict(value).items():
+                value[k] = parse_strings_with_macro_refs(v)
+        elif isinstance(value, list):
+            value = [parse_strings_with_macro_refs(v) for v in value]
+
+        return value
+
     for field_name, field_info in ModelMeta.all_field_infos().items():
         field = field_info.alias or field_name
+        field_value = fields.get(field)
 
-        if field in RUNTIME_RENDERED_MODEL_FIELDS:
+        if field in ("cron", "description") or field_value is None:
             continue
 
-        field_value = fields.get(field)
-        if field_value is None:
+        if field in RUNTIME_RENDERED_MODEL_FIELDS:
+            fields[field] = parse_strings_with_macro_refs(field_value)
```

Comment:
I'm somewhat confused by how these runtime rendered fields (signals, audits, etc) might end up being strings? Aren't they always expressions?

Reply:
The issue appears with python models, where you could encode values as strings that are later parsed during validation.